### PR TITLE
Exports Refactor Outline

### PR DESCRIPTION
**Pull request by NoahCarnahan**
@benrudolph here's the outline I've been playing around with, with a few tests demonstrating how I've been thinking the `TableConfiguration` will behave. I'm imagining that each `TableConfiguration` is configured by a user selecting some subset of the items in an `ExportableItems`.
So, the part I was thinking might be good for you to work on now is constructing `ExportableItems`s from apps.

### CODE REVIEW

File: corehq/apps/export/models.py [hunk 1]
```diff
@@ -8,6 +11,144 @@
 from corehq.apps.app_manager.models import Application
 from dimagi.utils.couch.database import iter_docs
 
+from corehq.apps.userreports import models
```
Comment (dimagimon) [inline corehq/apps/export/models.py]: F401 'models' imported but unused

File: corehq/apps/export/models.py [hunk 2]
```diff
@@ -8,6 +11,144 @@
...
+from dimagi.utils.decorators.memoized import memoized
```
Comment (dimagimon) [inline corehq/apps/export/models.py]: F401 'memoized' imported but unused

File: corehq/apps/export/models.py [hunk 3]
```diff
@@ -8,6 +11,144 @@
...
+        return rows
+
+    def _get_sub_documents(self, path, docs):
+        """
+        Return each instance of a repeat group at the path from the given docs.
+        If path is [], just return the docs
+
+        >>> TableConfiguration()._get_sub_documents(['foo'], [{'foo': {'bar': 'a'}}, {'foo': {'bar': 'b'}}])
+        [{'bar': 'a'}, {'bar': 'b'}]
+        >>> TableConfiguration()._get_sub_documents(['foo', 'bar'], [{'foo': [{'bar': {'baz': 'a'}}, {'bar': {'baz': 'b'}},]}]
```
Comment (dimagimon) [inline corehq/apps/export/models.py]: E501 line too long (126 > 115 characters)

**Comment by czue** [comment]
(Made some comments on https://github.com/dimagi/commcare-hq/pull/9985 - sorry wasn't sure how best to review this)

**Pull request closed without merging**